### PR TITLE
[RESTEASY-2201] update jackson in jose-jwt

### PR DESCRIPTION
**Pull request by asoldano**

### CODE REVIEW

**Comment by marekkopecky** [comment]
WF module.xml needs to be updated as well

File: testsuite/unit-tests/src/test/java/org/jboss/resteasy/test/jose/JWSTest.java [hunk 1]
```diff
@@ -47,8 +48,8 @@
    @Test
    public void testHeaderSerialization() throws Exception {
       ObjectMapper mapper = new ObjectMapper();
-      mapper.setSerializationInclusion(JsonSerialize.Inclusion.NON_NULL);
```
Comment (marekkopecky) [inline testsuite/unit-tests/src/test/java/org/jboss/resteasy/test/jose/JWSTest.java]: This is backward incompatible change,but jose-jwt module is private (see [module.xml](https://github.com/wildfly/wildfly/blob/master/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/resteasy/jose-jwt/main/module.xml#L28)), so WF users shouldn't use this directly.

Comment (asoldano) [reply testsuite/unit-tests/src/test/java/org/jboss/resteasy/test/jose/JWSTest.java]: Yep, exactly what I was thinking, but I'd like to have your (QE) go/no-go on this before merging.

**Comment by asoldano** [comment]
> WF module.xml needs to be updated as well
> 
> https://github.com/wildfly/wildfly/blob/master/feature-pack/src/main/resources/modules/system/layers/base/org/jboss/resteasy/jose-jwt/main/module.xml

That would be done when upgrading the resteasy version in wfly, yep

**Comment by asoldano** [comment]
Having heard no real objections, I'm merging this (cc @marekkopecky )

**Pull request merged**